upload/extension/opencart/catalog/model/payment/opayo.php

```diff
@@ -0,0 +1,707 @@
+			$subscription_end = new DateTime(date_format($trial_end, 'Y-m-d H:i:s'));
+			$subscription_end = $this->calculateSchedule($item['subscription']['frequency'], $subscription_end, $item['subscription']['cycle'] * $item['subscription']['duration']);
+		} elseif (date_format($trial_end, 'Y-m-d H:i:s') == date_format($subscription_end, 'Y-m-d H:i:s') && $item['subscription']['duration'] != 0) {
+			$next_payment = $this->calculateSchedule($item['subscription']['frequency'], $next_payment, $item['subscription']['cycle']);
+			$subscription_end = $this->calculateSchedule($item['subscription']['frequency'], $subscription_end, $item['subscription']['cycle'] * $item['subscription']['duration']);
+		} elseif (date_format($trial_end, 'Y-m-d H:i:s') > date_format($subscription_end, 'Y-m-d H:i:s') && $item['subscription']['duration'] == 0) {
+			$subscription_end = new DateTime('0000-00-00');
+		} elseif (date_format($trial_end, 'Y-m-d H:i:s') == date_format($subscription_end, 'Y-m-d H:i:s') && $item['subscription']['duration'] == 0) {
+			$next_payment = $this->calculateSchedule($item['subscription']['frequency'], $next_payment, $item['subscription']['cycle']);
+			$subscription_end = new DateTime('0000-00-00');
+		}
+
+		if (date_format($trial_end, 'Y-m-d H:i:s') >= date_format($subscription_end, 'Y-m-d H:i:s')) {
+			$recurring_expiry = date_format($trial_end, 'Y-m-d');
+		} else {
+			$recurring_expiry = date_format($subscription_end, 'Y-m-d');
+		}
+
+		$recurring_frequency = date_diff(new DateTime('now'), new DateTime(date_format($next_payment, 'Y-m-d H:i:s')))->days;
+
+		$response_data = $this->setPaymentData($order_info, $opayo_order_info, $price, $subscription_id, $item['subscription']['name'], $recurring_expiry, $recurring_frequency);
+
+		$this->addSubscriptionOrder($this->session->data['order_id'], $response_data, $subscription_id, date_format($trial_end, 'Y-m-d H:i:s'), date_format($subscription_end, 'Y-m-d H:i:s'));
+
+		if ($response_data['Status'] == 'OK') {
+			$this->updateSubscriptionOrder($subscription_id, date_format($next_payment, 'Y-m-d H:i:s'));
+
+			$this->addSubscriptionTransaction($subscription_id, $this->session->data['order_id'], $response_data, 1);
+		} else {
+			$this->addSubscriptionTransaction($subscription_id, $this->session->data['order_id'], $response_data, 4);
+		}
+	}
+
+	/**
+	 * Cron Payment
+	 *
+	 * @return array
+	 */
+	public function cronPayment(): array {
+		$this->load->model('account/order');
+
+		$subscriptions = $this->getProfiles();
+		$cron_data = [];
+		$i = 0;
+
+		foreach ($subscriptions as $subscription) {
+			$subscription_order = $this->getSubscriptionOrder($subscription['subscription_id']);
+
+			$today = new DateTime('now');
+			$unlimited = new DateTime('0000-00-00');
+			$next_payment = new DateTime($subscription_order['next_payment']);
+			$trial_end = new DateTime($subscription_order['trial_end']);
+			$subscription_end = new DateTime($subscription_order['subscription_end']);
```

Comment:
All of these uses of `DateTime` need to be written as `\DateTime` since you are in a indifferent namespace and OC doesn't import namespaces.